### PR TITLE
2024a version Package upgrades across all images.

### PR DESCRIPTION
**Pull request by dibryant**
Fixes for https://issues.redhat.com/browse/RHOAIENG-4245
## Description

Manually updated each python package to major version. 
## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by harshad16** [comment]
/hold

updating jupyterlab to 4.0.0+ version needs to be verified
Please other feel free to review, just don't merge it yet.

File: .github/workflows/piplock-renewal-2023b.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ jobs:
       - name: Checkout code from the release branch
         uses: actions/checkout@v3
         with:
-          ref: 2023b
+          ref: action
```
Comment (rkpattnaik780) [inline .github/workflows/piplock-renewal-2023b.yml:26]: We can remove this now that we got the updates generated.

**Comment by atheo89** [comment]
This PR updates the Python packages manually, instead, this should be done automatically though an action (according to issue 1855). Please consider the following:

- Updating the runtime images as well.
- Updating the test notebooks to point to the updated versions.
- Advising on the compatibility [matrix](https://github.com/jupyterlab/jupyterlab/issues/14590) of JupyterLab 4.0 to ensure the rest of the python packages are updated accordingly.
- Some packages probably should be pinned to specific versions, as we've done in the past.
- Checking the compatibility between CUDA and TensorFlow & PyTorch versions.
- Updating this [document](https://docs.google.com/spreadsheets/d/1LaTLTQSY9g8Rgau6jJwBQmZjd2NBfnElYZthfb9Bhwk/edit#gid=0) with the new Python package versions on the column 2024a.

@harshad16, seems that the scope of '1855' is to create a brand new action that updates the Python packages on our behalf. Should we consider to track this workload in a separate epic issue in case of delay?

**Comment by harshad16** [comment]
@atheo89 as you pointed out, this was to be done with automation.
However, we are pivoting a bit and now currently just concentration on doing manual updates, so we are ready 2024.a
This PR would be linked to this issue: https://issues.redhat.com/browse/RHOAIENG-4195 going forward.

File: jupyter/pytorch/ubi9-python-3.9/Pipfile [hunk 2]
```diff
@@ -12,44 +12,49 @@ name = "pytorch"
```
Comment (atheo89) [inline jupyter/pytorch/ubi9-python-3.9/Pipfile:12]: We need to update the source link from: `url = "https://download.pytorch.org/whl/cu121"` to  `url = "https://download.pytorch.org/whl/cu121"`

File: jupyter/trustyai/ubi8-python-3.8/Pipfile [hunk 3]
```diff
@@ -7,42 +7,47 @@ name = "pypi"
 
 [packages]
 # TrustyAI packages
-trustyai = "~=0.3.0"
+trustyai = "~=0.5.0"
```
Comment (atheo89) [inline jupyter/trustyai/ubi8-python-3.8/Pipfile:10]:
```suggestion
trustyai = "~=0.6.0"
```

There is a new version for TrustyAI

**Comment by harshad16** [comment]
/unhold
Made all the required changes
lets merge this in
and check for fix in separate PR.
/approve

**Pull request merged**